e2e_test/batch/basic/to_jsonb.slt.part

```diff
@@ -0,0 +1,249 @@
+----
+"-Infinity"
+
+query T
+SELECT to_jsonb(true);
+----
+true
+
+query T
+SELECT to_jsonb('2023-10-01'::date);
+----
+"2023-10-01"
+
+query T
+SELECT to_jsonb('10:23:45'::time);
+----
+"10:23:45"
+
+query T
+select to_jsonb('30 second'::interval);
+----
+"00:00:30"
+
+query T
+SELECT to_jsonb('2023-10-01 10:23:45'::timestamp);
+----
+"2023-10-01T10:23:45"
+
+query T
+SELECT to_jsonb('2014-05-28 12:22:35.614298'::timestamp);
+----
+"2014-05-28T12:22:35.614298"
+
+query T
+SELECT to_jsonb('2023-06-01 00:00:00Z'::timestamptz);
+----
+"2023-06-01T00:00:00+00:00"
+
+statement ok
+set timezone to 'UTC';
+
+query T
+SELECT to_jsonb('2014-05-28 12:22:35.614298'::timestamptz);
+----
+"2014-05-28T12:22:35.614298+00:00"
+
+statement ok
+set timezone to 'Europe/London';
+
+query T
+SELECT to_jsonb('2014-05-28 12:22:35.614298'::timestamptz);
+----
+"2014-05-28T11:22:35.614298+00:00"
+
+statement ok
+SET timezone = 'EST5EDT';
+
+query T
+SELECT to_jsonb('2014-05-28 12:22:35.614298'::timestamptz);
+----
+"2014-05-28T16:22:35.614298+00:00"
+
+statement ok
+set timezone to 'UTC';
+
+query T
+SELECT to_jsonb('2014-05-28 12:22:35.614298'::timestamptz);
+----
+"2014-05-28T12:22:35.614298+00:00"
+
+query T
+SELECT to_jsonb('{"key": "value"}'::jsonb);
+----
+{"key": "value"}
+
+query T
+select to_jsonb(null::varchar);
+----
+null
+
+query T
+SELECT to_jsonb(NULL::integer);
+----
+null
+
+query T
+SELECT to_jsonb(NULL::text);
+----
+null
+
+query T
+SELECT to_jsonb(''::text);
+----
+""
+
+query T
+select to_jsonb(ARRAY[]::integer[]);
+----
+[]
+
+query I
+select to_jsonb(1::bigint);
+----
+1.0
+
+query I
+select to_jsonb(1::bigint << 55);
+----
+3.602879701896397e16
+
+query I
+select to_jsonb(1::bigint << 63);
+----
+-9.223372036854776e18
+
+query T
+select to_jsonb(ARRAY[3,4,5,6]);
+----
+[3, 4, 5, 6]
+
+query T
+SELECT to_jsonb(ARRAY['apple', 'banana', 'cherry']);
+----
+["apple", "banana", "cherry"]
+
+query T
+select to_jsonb(row(row(1, 5), 'tojsonb', null));
+----
+{"f1": {"f1": 1, "f2": 5}, "f2": "tojsonb", "f3": null}
+
+query T
+select to_jsonb(row(row(1, 5)::struct<a int, b int>, 'tojsonb', null));
+----
+{"f1": {"a": 1, "b": 5}, "f2": "tojsonb", "f3": null}
+
+query T
+select to_jsonb(array[row(1, 't')::struct<a int, b varchar>, row(2, 's')::struct<a int, b varchar>, row(3, 'g')::struct<a int, b varchar>]);
+----
+[{"a": 1, "b": "t"}, {"a": 2, "b": "s"}, {"a": 3, "b": "g"}]
+
+query T
+select to_jsonb(row(row(1, array[1, 3, 4], row(45, 'to_char', 'to_date', array['2023-10-10 10:55:45'::timestamp, '2023-10-01 10:23:45'::timestamp])), 'tojsonb', null));
+----
+{"f1": {"f1": 1, "f2": [1, 3, 4], "f3": {"f1": 45, "f2": "to_char", "f3": "to_date", "f4": ["2023-10-10T10:55:45", "2023-10-01T10:23:45"]}}, "f2": "tojsonb", "f3": null}
+
+statement ok
+CREATE TABLE structtype (st struct<key int, val varchar>);
+
+statement ok
+INSERT INTO structtype VALUES (row(6, 'v1')::struct<key int, val varchar>), (row(7, 'v2')::struct<key int, val varchar>);
```

Comment:
nit: explicit cast is unnecessary during insert
```suggestion
INSERT INTO structtype VALUES (row(6, 'v1')), (row(7, 'v2'));
```

Reply:
updated